src/views/OnTouchGui.jsx

```diff
@@ -89,120 +91,102 @@ class OnTouchGui extends Component {
-    if (storyIdentifierNode.Value !== selectedStory) {
-      const json = this.addStoryTree(selectedStory);
+    const previousStory = storyIdentifierNode.value;
+    this.setState({ currentStory: selectedStory });
 
-      // If the previous story was the default there is no tags to remove
-      if (storyIdentifierNode.Value !== defaultStory) {
-        this.props.ChangePropertyValue(applyRemoveTag.Description, '');
-      }
+    // Return if the selected story is the same as the OpenSpace property value
+    if (previousStory === selectedStory)
+      return;
 
-      // If the current story is the default - hide distance info on screen
-      if (selectedStory === defaultStory) {
-        showDistanceOnScreen(false);
-      }
-      else {
-        showDistanceOnScreen(true);
-      }
+    // If the previous story had an empty focus node list
+    if (focusNodesList.value.length !== 0) {
+      const focusNodes = focusNodesList.value.split(',');
+      removeStoryTags(this.props.luaApi, focusNodes);
+    }
 
-      // Set all the story specific properties
-      this.props.ChangePropertyValue(storyIdentifierNode.Description, selectedStory);
-      this.props.ChangePropertyValue(focusNodesList.Description, json.focusbuttons);
-      applyStoryInterestingTagsToFocusNodes();
-      this.props.ChangePropertyValue(applyAddTag.Description, '');
-      this.props.ChangePropertyValue(anchorNode.Description, json.start.planet);
-      this.props.ChangePropertyValue(overViewNode.Description, json.overviewlimit);
-      setStoryStart(json.start.location, json.start.time);
-
-      // Check settings of the previous story and reset values
-      this.checkStorySettings(this.props.story, 'true');
-      // Check and set the settings of the current story
-      this.checkStorySettings(json, 'false');
-
-      // If the previous story scaled planets reset value
-      if (this.props.story.scaleplanets) {
-        this.props.scaleNodes.forEach((planet) => {
-          this.props.ChangePropertyValue(planet.Description, 1);
-        });
-      }
-      // If the previous story toggled bool properties reset them to default value
-      if (this.props.story.toggleboolproperties) {
-        this.props.story.toggleboolproperties.forEach((property) => {
-          resetBoolProperty(property.URI, property.defaultvalue);
-        });
-      }
+    const json = this.addStoryTree(selectedStory);
+    
+    // If the current story is the default - hide distance info on screen
+    if (selectedStory === DefaultStory) {
+      showDistanceOnScreen(this.props.luaApi, false);
     }
+    else {
+      showDistanceOnScreen(this.props.luaApi, true);
+    }
+
+    // Set all the story specific properties
+    this.props.changePropertyValue(storyIdentifierNode.description.Identifier, selectedStory);
+    if(json.focusbuttons){
+      this.props.changePropertyValue(focusNodesList.description.Identifier, json.focusbuttons.toString());
```

Comment:
`toString()` could be replaced with `.join(',')`

Reply:
...or maybe you also need to add quotes then... like `'"' + ` ... `.join('","')` ...  ` + '"'`